swagger_server/configuration.py

```diff
@@ -0,0 +1,141 @@
+from yaml import load, dump
+from json import loads
+from os import getenv
+from pathlib import Path
+from sqlalchemy import create_engine, event, DDL
+from sqlalchemy.ext.declarative import declarative_base
+from sqlalchemy.orm import scoped_session, sessionmaker
+from sqlalchemy_utils import database_exists, create_database
+from .controllers_local.encryption import ONSCryptographer
+
+class ONSEnvironment(object):
+
+    def __init__(self):
+        """
+        Nothing actually happens at this point, we're just setting up variables
+        for future reference. In particular _base and _env are critical to the
+        setup of other components.
+        """
+        self._port = 0
+        self._crypto_key = None
+        self._ons_cipher = None
+        self._config = ConfigParser()
+        self._config._interpolation = ExtendedInterpolation()
+        self._config.read('config.ini')
+        self._env = getenv('ONS_ENV', 'development')
+        self._session = scoped_session(sessionmaker())
+        self._base = declarative_base()
+        #
+        #   If we're using postgres, make sure we create the appropriate Schema if it
+        #   doesn't already exist.
+        #
+        schema = self.get('db_schema')
+        if schema:
+            event.listen(
+                self._base.metadata,
+                "before_create",
+                DDL('CREATE SCHEMA IF NOT EXISTS {}'.format(schema)).execute_if(dialect='postgresql')
+            )
+        self._engine = create_engine(self.get('db_connection'), convert_unicode=True)
+
+    def activate(self):
+        """
+        Activate the database, then set up the Cloud Foundry environment if there is one ...
+        """
+        self._activate_database()
+        self._activate_cf()
+        self._ons_cipher = ONSCryptographer(self._crypto_key)
+
+    def _activate_database(self):
+        """
+        Connect to the database (create it if it's missing) and set up tables as per our models.
+        If we're in a 'test' environment, drop all the tables first ...
+        """
+        self._session.remove()
+        self._session.configure(bind=self._engine, autoflush=False, autocommit=False, expire_on_commit=False)
+        if not database_exists(self._engine.url):
+            create_database(self._engine.url)
+        if self.if_drop_database:
+            self._base.metadata.drop_all(self._engine)
+        from .models_local import _models
+        self._base.metadata.create_all(self._engine)
+
+    def _activate_cf(self):
+        """
+        If we're running on Cloud Foundry we need to set the host in the swagger YAML file to be
+        the hostname we recover from VCAP application. We also need to set the configuration port for
+        use by the listener if we're running locally.
+        """
+        self._crypto_key = getenv('ONS_CRYPTOKEY', self.get('crypto_key'))
+        config = './swagger_server/swagger/swagger.yaml'
```

Comment:
Yes it could go in config.ini, but the location is set by the code generator, i.e. it's a constant and config.ini is designed to track things that could change so they can be altered without altering the code. If we do this there are a few other instances that should also be refactored, but again this is all codegen stuff and not project specific. Would be good to get issues logged on the Codegen repo ..